nodl_python/nodl/_parsing/_v1/_parsing.py

```diff
@@ -0,0 +1,131 @@
+# Copyright 2020 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify it under the terms of the
+# GNU Limited General Public License version 3, as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+# without even the implied warranties of MERCHANTABILITY, SATISFACTORY QUALITY, or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Limited General Public License for more details.
+#
+# You should have received a copy of the GNU Limited General Public License along with
+# this program. If not, see <http://www.gnu.org/licenses/>.
+
+from typing import List
+import warnings
+
+from lxml import etree
+from nodl._parsing._qos import parse_qos
+from nodl._util import get_bool_attribute
+from nodl.exception import InvalidNoDLException
+from nodl.types import Action, Node, Parameter, Service, Topic
+from nodl.warning import NoNodeInterfaceWarning
+
+
+def parse_action(element: etree._Element) -> Action:
```

Comment:
This function is internal to this module. No one else (other than tests) use it. Consider prefixing with an underscore, and consider the same for other functions throughout.